odoo/api.py

```diff
@@ -1037,10 +1034,11 @@ def insert_missing(self, records, field, values):
         existing values in cache.
         """
         field_cache = self._set_field_cache(records, field)
+        env = records.env
         if field.translate:
-            if records.env.context.get('prefetch_langs'):
-                langs = {lang for lang, _ in records.env['res.lang'].get_installed()} | {'en_US'}
-                _langs = {f'_{l}' for l in langs} if field._lang(records.env).startswith('_') else set()
+            if env.context.get('prefetch_langs'):
+                langs = OrderedSet(lang for lang, _ in env['res.lang'].get_installed()) | OrderedSet(('en_US', env.lang))
+                _langs = OrderedSet(f'_{l}' for l in langs) if field._lang(env).startswith('_') else OrderedSet()
```

Comment:
Seeing the code below, we can say that `langs` should be a set (if large), but `_langs` shouldn't.
```suggestion
                installed = [lang for lang, _ in env['res.lang'].get_installed()]
                langs = OrderedSet(installed + ['en_US', env.lang])
                _langs = [f'_{l}' for l in langs] if field._lang(env).startswith('_') else []
```